### PR TITLE
use proper bibliography with cross-links

### LINKED ISSUE #16: Update/Fix bibliography
As the current bibliography does not work at all, it has to be fixed. You can use the updated version of the template, which explains in detail how to properly use the bibliography and its labels. 

In addition, instead of having a bullet list in each chapter with references to the used sources, a comma separated list is way more beautiful.

### PR DESCRIPTION
**Pull request by larsrh**
Resolves #16 

This PR does not add more references than were there before. Consequently, there's no reference section in chapters 4 and 5. Those can be added in a future version of the module.

### CODE REVIEW

File: docs/02-module-contracts/references.adoc [hunk 1]
```diff
@@ -1,4 +1,3 @@
 === {references}
 
-* Andreas Antonopoulos, Gavin Wood: "Mastering Ethereum: Building Smart Contracts and Dapps"
-* Consensys: "Ethereum Smart Contract Best Practices, https://consensys.github.io/smart-contract-best-practices/known_attacks/
+<<antonopoulos2>>, <<consensys>>
```
Comment (programming-wolf) [inline docs/02-module-contracts/references.adoc]: You cannot have digits in labels, unfortunately. :-/

File: docs/99-references/00-references.adoc [hunk 2]
```diff
@@ -13,20 +13,49 @@ Dieser Abschnitt enthält Quellenangaben, die ganz oder teilweise im Curriculum
 This section contains references that are cited in the curriculum.
 // end::EN[]
```
Comment (programming-wolf) [inline docs/99-references/00-references.adoc:14]: You have so many references but you barely reference any of them.

**Review by programming-wolf — CHANGES_REQUESTED** [review]

File: docs/99-references/00-references.adoc (hunk 2 above)
Comment (larsrh) [reply docs/99-references/00-references.adoc:14]: I'm aware, but IMHO rectifying this would warrant a new version of the module.

Comment (programming-wolf) [reply docs/99-references/00-references.adoc:14]: Fair enough.

**Review by programming-wolf — APPROVED** [review]
Better, thanks.

**Pull request merged**